tesla_ver/bubble_chart.py

```diff
@@ -162,31 +183,24 @@ def update_figure(
         ]:
             raise PreventUpdate
 
-        df = pd.read_json(json_data)
-        traces = list()
-
-        # filtered_df contains only the X values from the time point specified by the slider
-        filtered_df = df[df["X"] == time_value].convert_dtypes()
-
-        # Using "alpha-3" as a unique identifier, the loop then creates a new trace for each entity defined by "alpha-3"
-        # and then appends it to the list of traces that defines the data for the figure to be graphed.
-        for entity in filtered_df["alpha-3"].unique():
-            df_by_value = filtered_df[filtered_df["alpha-3"] == entity]
-            traces.append(
-                Scattergl(
-                    x=df_by_value[x_column_name],
-                    y=df_by_value[y_column_name],
-                    mode="markers",
-                    opacity=0.9,
-                    name=entity,
-                    hovertext=df_by_value[annotation_column_name].values.tolist(),
-                ),
-            )
+        df_by_time = pd.DataFrame.from_dict(ast.literal_eval(json.loads(json_data).get(str(time_value))))
```

Comment:
Add a comment on the operations being performed here

Reply:
addressed in 6e8fe20